src/details/ArborX_DetailsUtils.hpp

```diff
@@ -321,13 +320,26 @@ minMax(ExecutionSpace &&space, ViewType const &v)
   static_assert(ViewType::rank == 1, "minMax requires a View of rank 1");
   auto const n = v.extent(0);
   ARBORX_ASSERT(n > 0);
-  Kokkos::MinMaxScalar<typename ViewType::non_const_value_type> result;
-  Kokkos::MinMax<typename ViewType::non_const_value_type> reducer(result);
+  using ValueType = typename ViewType::non_const_value_type;
+  ValueType min_val;
+  ValueType max_val;
   Kokkos::RangePolicy<std::decay_t<ExecutionSpace>> policy(
       std::forward<ExecutionSpace>(space), 0, n);
-  Kokkos::parallel_reduce("ArborX::Algorithms::minmax", policy,
-                          Kokkos::Impl::min_max_functor<ViewType>(v), reducer);
-  return std::make_pair(result.min_val, result.max_val);
+  Kokkos::parallel_reduce(
+      "ArborX::Algorithms::minmax", policy,
+      KOKKOS_LAMBDA(int i, ValueType &local_min, ValueType &local_max) {
+        auto const &val = v(i);
+        if (val < local_min)
+        {
+          local_min = val;
+        }
+        if (val > local_max)
+        {
+          local_max = val;
+        }
+      },
+      Kokkos::Min<ValueType>(min_val), Kokkos::Max<ValueType>(max_val));
```

Comment:
I did not realize you can do multiple join operations for a reducer. Is that new?

Reply:
I commented about it in the description.  It is there since Kokkos 3.2

Reply:
https://github.com/kokkos/kokkos/pull/3052. We should have that since 3.2.